contrib/format-hdf5/src/main/java/org/apache/drill/exec/store/hdf5/writers/HDF5DoubleDataWriter.java

```diff
@@ -0,0 +1,103 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.drill.exec.store.hdf5.writers;
+
+import ch.systemsx.cisd.hdf5.IHDF5Reader;
+import org.apache.drill.common.types.TypeProtos;
+import org.apache.drill.exec.physical.resultSet.RowSetLoader;
+import org.apache.drill.exec.store.hdf5.HDF5Utils;
+import org.apache.drill.exec.vector.accessor.ScalarWriter;
+
+import java.util.List;
+
+public class HDF5DoubleDataWriter extends HDF5DataWriter {
+
+  private final double[] data;
+
+  private final ScalarWriter rowWriter;
+
+  // This constructor is used when the data is a 1D column.  The column is inferred from the datapath
+  public HDF5DoubleDataWriter(IHDF5Reader reader, RowSetLoader columnWriter, String datapath) {
+    super(reader, columnWriter, datapath);
+    data = reader.readDoubleArray(datapath);
+
+    fieldName = HDF5Utils.getNameFromPath(datapath);
+    rowWriter = makeWriter(columnWriter, fieldName, TypeProtos.MinorType.FLOAT8, TypeProtos.DataMode.OPTIONAL);
+  }
+
+  // This constructor is used when the data is part of a 2D array.  In this case the column name is provided in the constructor
+  public HDF5DoubleDataWriter(IHDF5Reader reader, RowSetLoader columnWriter, String datapath, String fieldName, int currentColumn) {
+    super(reader, columnWriter, datapath, fieldName, currentColumn);
+    // Get dimensions
+    long[] dimensions = reader.object().getDataSetInformation(datapath).getDimensions();
+    double[][] tempData;
+    if (dimensions.length == 2) {
+      tempData = transpose(reader.readDoubleMatrix(datapath));
+    } else {
+      tempData = transpose(reader.float64().readMDArray(datapath).toMatrix());
+    }
+    data = tempData[currentColumn];
+    rowWriter = makeWriter(columnWriter, fieldName, TypeProtos.MinorType.FLOAT8, TypeProtos.DataMode.OPTIONAL);
+  }
+
+  public HDF5DoubleDataWriter(IHDF5Reader reader, RowSetLoader columnWriter, String fieldName, List tempListData) {
+    super(reader, columnWriter, null);
+    this.fieldName = fieldName;
+    data = new double[tempListData.size()];
+    for (int i = 0; i < tempListData.size(); i++) {
+      data[i] = (Double)tempListData.get(i);
+    }
+    rowWriter = makeWriter(columnWriter, fieldName, TypeProtos.MinorType.FLOAT8, TypeProtos.DataMode.OPTIONAL);
+  }
+
+
+  public boolean write() {
+    if (counter > data.length) {
+      return false;
+    } else {
+      rowWriter.setDouble(data[counter++]);
+      return true;
+    }
+  }
+
+  public boolean hasNext() {
+    return counter < data.length;
+  }
+
+  private double[][] transpose(double[][] array) {
```

Comment:
Please add java doc here to explain logic of this method.

Reply:
Hi Arina, 
Basically, I needed a way to transpose primitive 2D arrays.   Since they are primitives, I implemented a version of this with the differnet data types for each primitive.  Is there a way to do this in a generic way for all primitive types?

Reply:
I couldn't figure any way to do this in a generic way, however, I'm not an expert on some of Java's tricks for generic data types.